lib/engine/share_pool.rb

```diff
@@ -29,7 +29,8 @@ def owner
       nil
     end
 
-    def buy_shares(entity, shares, exchange: nil, exchange_price: nil, swap: nil, allow_president_change: true, silent: nil)
+    def buy_shares(entity, shares, exchange: nil, exchange_price: nil, swap: nil,
```

Comment:
What does the `borrow_from?` parameter mean here?

Reply:
It's a new feature added by this PR to borrow money from another player when purchasing this share.

So the action is Player A buys a share with borrow_from: Player B. This means:
- During the check for proper cash, the cache of both players is combined
- When purchasing if Player A doesn't have enough, money is spend from Player B to make up the difference

1871 uses this mechanic for Union Bank purchasing.